### ISSUE TITLE
How to handle duplicate keys?

### ISSUE THREAD

**Issue by Professor-Heavy** [shravan2x/Gameloop.Vdf#8]
Sorry for the silly question, I'm new to this entire C# scene, and I'm using this parser for a project. As a result, I'm probably being a complete and total idiot and that my issue is common C# knowledge.

If two keys with the same name exist in the same object, they're both used in that object but have the same value. In the VDF formatted file that I'm using, though, they both have different values.

In other words, is it possible to access these values like a numeric array to avoid this issue? Or is there anything else in this parser that lets me do this?

**Comment by shravan2x** [comment]
No, your question is definitely a good one. I'll include this in the FAQ at some point.

In general, you cannot access values of duplicate keys in the same way you would a dictionary, which `VObject` [inherits from](https://github.com/shravan2x/Gameloop.Vdf/blob/master/Gameloop.Vdf/Linq/VObject.cs#L10). So when you access a property by its key name, you always get the first corresponding value.

But there is something you can do. `VObject` maintains a copy of every child `VProperty`, which you can access by calling `Children()`. Assume you have VDF data like:

```
"root"
{
    "key1" "value1"
    "key2" "value2"
    "key1" "value3"
}
```

To access the last value of "key1", just do:
```cs
VProperty root = VdfConvert.Deserialize(vdf);
Console.WriteLine(root.Value["key1"]); // Prints 'value1'
Console.WriteLine(root.Value["key2"]); // Prints 'value2'
Console.WriteLine(root.Value.Children().Last(x => x.Key == "key1").Value); // Prints 'value3'
```

As an aside, could you tell me where you found the erroneous VDF data you have? Did you find it in the Steam installation directory?

**Comment by shravan2x** [comment]
To add to my previous response, if you need an array of values, just do:

```
VToken[] vals = root.Value.Children().Where(x => x.Key == "key1").Select(x => x.Value).ToArray();
```

**Comment by shravan2x** [comment]
I'm closing this issue due to inactivity. If you have further questions, feel free to reopen.

**Comment by Bluscream** [comment]
If anyone else stumbles across this issue and just wants to treat duplicates differently:

```c#
var json = VdfConvert.Deserialize(text).ToJson(new VdfJsonConversionSettings() { ObjectDuplicateKeyHandling = DuplicateKeyHandling.Ignore, ValueDuplicateKeyHandling = DuplicateKeyHandling.Ignore });
```

**Comment by shravan2x** [comment]
@Bluscream Note that your solution only applies when converting a `VObject` to a `JObject`. I believe the author was asking about accessing properties with duplicate keys directly within a `VObject`.

**Comment by ZzZombo** [comment]
There is one misconception about VDF: **it's not a dictionary**! The best unambiguous way to represent parsed VDF data is to use a list.
